web/src/stream_data.ts

```diff
@@ -579,36 +579,17 @@ export function can_preview(sub: StreamSubscription): boolean {
     return sub.subscribed || !sub.invite_only || sub.previously_subscribed;
 }
 
-export function can_change_permissions(sub: StreamSubscription): boolean {
-    // Whether the current user has permission to administer this stream.
-    // Organisation admins have this permission regardless of whether
-    // they are part of can_administer_channel_group. Non-subscribers with
-    // these permission can edit name and description of a private channel
-    // without being subscribed to it.
-
-    if (sub.invite_only && !sub.subscribed) {
+export function can_change_permissions(
+    sub: StreamSubscription,
+    require_content_access = true,
+): boolean {
+    if (require_content_access && !has_content_access(sub)) {
+        return false;
+    } else if (!require_content_access && !has_metadata_access(sub)) {
         return false;
     }
```

Comment:
It might be preferable to just have two functions: `can_change_metadata_access_permissions` and `can_change_content_access_permissions`. Not sure, just thinking it might feel more readable.

Reply:
Oh, I see you rename them a few commits later. I think your names are probably better than what I suggested here. We can always refactor later if we come up with better naming, but given they're only called in a half-dozen places, there's not much to gain from having shorter, less explicit names.